specs/core/0_beacon-chain.md

```diff
@@ -1947,13 +1965,13 @@ def process_penalties_and_exits(state: BeaconState) -> None:
             break
 ```
 
-### Final updates
+#### Final updates
 
 * Set `state.latest_penalized_balances[(next_epoch) % LATEST_PENALIZED_EXIT_LENGTH] = state.latest_penalized_balances[current_epoch % LATEST_PENALIZED_EXIT_LENGTH]`.
 * Set `state.latest_randao_mixes[next_epoch % LATEST_RANDAO_MIXES_LENGTH] = get_randao_mix(state, current_epoch)`.
 * Remove any `attestation` in `state.latest_attestations` such that `slot_to_epoch(attestation.data.slot) < current_epoch`.
 
-## State root processing
+### State root verification
```

Comment:
It's tempting to put this at the end of per-block processing, .e.g. "Verify `block.state_root == hash_tree_root(state)`   after the per-epoch processing below."

Reply:
I see the reasoning, but at the end of the day, the implementation will have it here (after the epoch processing) so I'd lean toward keeping as is.